### ISSUE TITLE
Integrate CurriculumScheduler (objective schedules) with HF Trainer

### ISSUE THREAD

**Issue by Lev-Stambler** [DeepOpt-com/WrinkleFree-Monorepo#63]
## Problem

The curriculum scheduler for objective schedules (`CurriculumScheduler`) was implemented but never integrated with the HF Trainer path. This feature allows:

- Phase-based training with different objective weights at different stages
- Smooth weight transitions (linear/cosine interpolation between phases)
- Subweight scheduling for distillation components (hidden/logits/attention balance over time)
- Data configuration switching between phases

## Current State

- `CurriculumScheduler` exists in `packages/training/src/wf_train/objectives/manager.py` (lines 88-335)
- `create_curriculum_scheduler()` factory exists in `packages/training/src/wf_train/objectives/factory.py`
- Config support exists (e.g., `configs/training/mixed_precision_distill.yaml`)
- **NOT wired up** to `train_hf.py` or `WrinkleFreeTrainer`

## Desired Behavior

1. Parse curriculum config in `train_hf.py`
2. Pass `CurriculumScheduler` to `ObjectiveManager`
3. Call `curriculum.step()` each training step (likely in a callback or trainer override)
4. Log curriculum weights to WandB (`schedule/{name}_weight`, `schedule/{name}_{sub}_subweight`)

## Key Files

| File | Purpose |
|------|---------|
| `packages/training/scripts/train_hf.py` | Main entry - needs curriculum init |
| `packages/training/src/wf_train/hf/trainer.py` | WrinkleFreeTrainer - needs to call scheduler |
| `packages/training/src/wf_train/objectives/manager.py` | CurriculumScheduler implementation |
| `packages/training/src/wf_train/objectives/factory.py` | `create_curriculum_scheduler()` |

## Example Config (already supported)

```yaml
curriculum:
  enabled: true
  interpolation: linear
  phases:
    - name: warmup
      end_ratio: 0.30
      objectives:
        distill: 1.0
      objective_subweights:
        distill:
          hidden: 0.0
          logits: 1.0
          attention: 1.0
    - name: main
      end_ratio: 1.0
      objectives:
        distill: 0.5
        ce: 0.5
```
